src/Compilers/CSharp/Portable/Compilation/CSharpCompilerDiagnosticAnalyzer.cs

```diff
@@ -27,19 +30,29 @@ internal override CommonMessageProvider MessageProvider
 
         internal override ImmutableArray<int> GetSupportedErrorCodes()
         {
-            var errorCodes = Enum.GetValues(typeof(ErrorCode));
-            var builder = ArrayBuilder<int>.GetInstance(errorCodes.Length);
-            foreach (ErrorCode errorCode in errorCodes)
+            var current = _supportedErrorCodes;
+            if (current.IsDefault)
```

Comment:
💡 Consider using `ImmutableInterlocked.InterlockedInitialize` here, provided the callback can be `static`

Reply:
sure :)